src/models/converters/data_decoded.rs

```diff
@@ -81,12 +74,12 @@ impl DataDecoded {
     pub(super) async fn build_address_info_index(
         &self,
         info_provider: &impl InfoProvider,
-    ) -> Option<HashMap<String, AddressInfo>> {
+    ) -> Option<HashMap<String, AddressEx>> {
         if !feature_flag_nested_decoding() {
             return None;
         }
 
-        let mut index = HashMap::new();
+        let mut index: HashMap<String, AddressEx> = HashMap::new();
```

Comment:
I think we can change this to a `Vec<AddressEx>`

Reply:
I mean the field `address_info_index`

Reply:
This would make the lookup more complicated, also the guarantee that you have no duplicates would not be given anymore